### PR TITLE
003 login system

### PR DESCRIPTION
**Pull request by AdamB-Kainos**
finished login system backend

### CODE REVIEW

File: src/main/java/org/example/TestApplication.java [hunk 1]
```diff
@@ -26,7 +34,12 @@ protected SwaggerBundleConfiguration getSwaggerBundleConfiguration(
     }
     @Override
     public void run(final TestConfiguration configuration,
-                    final Environment environment) {
+                    final Environment environment) throws SQLException {
```
Comment (ThomasS-kainos) [inline src/main/java/org/example/TestApplication.java]: Is the SQL Exception not caught sooner?

File: src/main/java/org/example/daos/AuthDao.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                            resultSet.getString("Hash"),
+                            resultSet.getInt("RoleId"));
+                } else {
+                    return null;  //user found but wrong password
+                }
+            }
+        throw new DoesNotExistException(Entity.USER); //user not found
+    }
+
+    public void generateUsers(final Connection c) throws SQLException,
```
Comment (ThomasS-kainos) [inline src/main/java/org/example/daos/AuthDao.java:48]: Is this needed in the live version, This is test code isn't it?

Comment (AdamB-Kainos) [reply src/main/java/org/example/daos/AuthDao.java:48]: this is needed

File: src/main/java/org/example/TestApplication.java (hunk 1 above)
Comment (AdamB-Kainos) [reply src/main/java/org/example/TestApplication.java]: fixed now

**Review by ThomasS-kainos — APPROVED** [review]

File: src/main/java/org/example/models/UserRole.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package org.example.models;
+
+import java.util.Map;
+
+public class UserRole {
+    public static final String ADMIN = "admin";
+    public static final String USER = "USER";
+    int roleId;
```
Comment (shaunganley) [inline src/main/java/org/example/models/UserRole.java]: why are these all different? make them consistent

File: src/test/java/org/example/integration/AuthIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class AuthIntegrationTest {
+    private static final DropwizardAppExtension<TestConfiguration> APP =
+            new DropwizardAppExtension<>(TestApplication.class);
+
+    @Test
+    void login_shouldLogUserIn() {
+        LoginRequest loginRequest = new LoginRequest(
+                "adam@random.com",
+                "pass123"
```
Comment (shaunganley) [inline src/test/java/org/example/integration/AuthIntegrationTest.java]: don't put valid creds in code, pass these in as env vars

File: src/test/java/org/example/services/LoginServiceTests.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    AuthService authService = new AuthService(jwtKey, authDao, databaseConnector);
+
+    LoginRequest loginRequest = new LoginRequest(
+            "adam@random.com",
+            "pass123"
+    );
+
+    Connection conn;
+
+    @Test
```
Comment (shaunganley) [inline src/test/java/org/example/services/LoginServiceTests.java:45]: you don't have a test for successful call?

**Review by shaunganley — CHANGES_REQUESTED** [review]
missing controller tests

**Comment by AdamB-Kainos** [comment]
> missing controller tests

**Comment by AdamB-Kainos** [comment]
> missing controller tests

resolved

**Comment by AdamB-Kainos** [comment]
> > missing controller tests
> 
> resolved

**Review by shaunganley — APPROVED** [review]

**Pull request merged**